src/order/succ_pred/relation.lean

```diff
@@ -0,0 +1,123 @@
+/-
+Copyright (c) 2022 Floris van Doorn. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Floris van Doorn
+-/
+import order.succ_pred.basic
+/-!
+# Relations on types with a `succ_order`
+
+This file contains properties about relations on types with a `succ_order`
+and their closure operations (like the transitive closure).
+-/
+
+open set relation succ_order pred_order function
+
+section succ
+variables {α : Type*} [linear_order α] [succ_order α] [is_succ_archimedean α]
+
+/-- For `n ≤ m`, `(n, m)` is in the reflexive-transitive closure of `~` if `i ~ succ i`
+  for all `i` between `n` and `m`. -/
+lemma refl_trans_gen_of_succ_of_le (r : α → α → Prop) {n m : α}
+  (h : ∀ i ∈ Ico n m, r i (succ i)) (hnm : n ≤ m) : refl_trans_gen r n m :=
```

Comment:
I find that it's often useful to unfold interval membership.
```suggestion
  (h : ∀ i, n ≤ i → i < m → r i (succ i)) (hnm : n ≤ m) : refl_trans_gen r n m :=
```
Maybe we should make that part of the simp normal form?

Reply:
I'm not sure if I agree that your suggestion is better, so I prefer to keep it the way it is.